File: 1260.py
# 4 5 1
# 1 2
# 1 3
# 1 4
# 2 4
# 3 4

from collections import deque

# dfs


def dfs(start):
    visited[start] = True
    print(start, end=" ")

    for i in graph[start]:
        if not visited[i]:
            dfs(i)

# bfs


def bfs(start):
    queue = deque([start])
    visited[start] = True
    while queue:
        v = queue.popleft()
        print(v, end=" ")
        for i in graph[v]:
            if not visited[i]:
                visited[i] = True
                queue.append(i)


n, m, v = map(int, input().split())
graph = [[]for _ in range(n+1)]

for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

# 정렬
for i in graph:
    i.sort()

# dfs
visited = [False]*(n+1)
dfs(v)
print()

# bfs
visited = [False]*(n+1)
bfs(v)

File: javascript/1260.js
const fs = require("fs");

const io = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = fs.readFileSync(io).toString();

const [metadata, ...edges] = input.trim().split("\n");
console.log(edges);

let [n, m, start] = metadata.split(" ").map(Number);

let edge = Array(n + 1)
  .fill()
  .map(() => []);

for (const e of edges) {
  let [from, to] = e.split(" ").map(Number);
  edge[from].push(to);
  edge[to].push(from);
}

let isVisitDFS = Array(n + 1).fill(false);
let isVisitBFS = Array(n + 1).fill(false);

let dfsResult = [];
let bfsResult = [];

function dfs(node) {
  dfsResult.push(node);
  isVisitDFS[node] = true;

  for (const n of edge[node]) {
    if (!isVisitDFS[n]) {
      dfs(n);
    }
  }
}

function bfs(node) {
  let queue = Array();

  queue.push(node);
  isVisitBFS[node] = true;
  bfsResult.push(node);

  while (queue.length > 0) {
    let now = queue.shift();

    for (const n of edge[now]) {
      if (!isVisitBFS[n]) {
        queue.push(n);
        isVisitBFS[n] = true;
        bfsResult.push(n);
      }
    }
  }
}

dfs(start);
bfs(start);

console.log(dfsResult.join(" "));
console.log(bfsResult.join(" "));
